pkg/noderesourcetopology/pluginhelpers.go

```diff
@@ -43,32 +45,26 @@ const (
 )
 
 func initNodeTopologyInformer(tcfg *apiconfig.NodeResourceTopologyMatchArgs, handle framework.Handle) (nrtcache.Interface, error) {
-	topoClient, err := topoclientset.NewForConfig(handle.KubeConfig())
+	scheme := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
```

Comment:
I would handle this error instead of panic

Reply:
this error in practice won't happen IMO. If we treat it as an error, @Tal-or do you mean we can find a way to proceed?

Reply:
@Huang-Wei Not sure if it's possible to proceed, but at least throwing a detailed error which bubble up and explained what happen, would be more user friendly.

Reply:
This is how the controller runtime client can know what to do with the types, this is a common usage from kube-builder, which is one of the most famous libs using controller runtime.

you can check: https://book.kubebuilder.io/cronjob-tutorial/empty-main

![image](https://github.com/kubernetes-sigs/scheduler-plugins/assets/3764335/a36a245a-f03d-4a7d-8b4b-30aeab78cdc3)

Reply:
My 2 cents is still that this error is not recoverable, and based on my exp. of writing controllers, this doesn't ever happen at all (pretty static behavior IMHO). So I'm more skewed with this pattern of panic immediately- if the error really happens.

Reply:
I agree with @Huang-Wei . The error is not recoverable, so it's fine to use `utilruntime.Must` and fail badly in the (I reckon) exceptionally rare case on which AddToScheme fails.

The only thing which doesn't fully click to me is why we need to fix the schema _in this precise code location_. Disclosure: I didn't review the latest changes, so this is possibly already improved.

Reply:
Hi @ffromani, I did not update the codes here.

we must handle the scheme somewhere, as the main scheduler, we added in main.go, and for this nrt plugin, I add it here when initing it, so that we are not adding this scheme if nrt plugin is not used.

as for `utilruntime.Must`, WDYT @Tal-or ?

Reply:
@ffromani raised a very good point. The AddToScheme logic should only happen once per its usage:

- for running scheduler, it's should in main's init() or equivalent place. @zwpaper I know this is generic code problem, but if you can add this logic in this PR, it'd be much appreciated 🙇‍♂️ 
- for UTs, it's fine to init in each UT as we treat each UT as a client

Reply:
> as for utilruntime.Must, WDYT @Tal-or ?
@zwpaper since the other reviewers are in favor of `utilruntime.Must` that's ok

Reply:
@Huang-Wei moved all scheme initial into `init` func, PTAL

```console
 rg -B 1 -g '!vendor/' -g '!*_test.go' -- utilruntime.Must
pkg/noderesourcetopology/plugin.go
45-func init() {
46:	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
47:	utilruntime.Must(topologyv1alpha2.AddToScheme(scheme))

pkg/networkaware/topologicalsort/topologicalsort.go
45-func init() {
46:	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
47:	utilruntime.Must(agv1alpha.AddToScheme(scheme))

pkg/networkaware/networkoverhead/networkoverhead.go
66-func init() {
67:	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
68-
69:	utilruntime.Must(agv1alpha1.AddToScheme(scheme))
70:	utilruntime.Must(ntv1alpha1.AddToScheme(scheme))

pkg/generated/clientset/versioned/scheme/register.go
54-	v1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
55:	utilruntime.Must(AddToScheme(Scheme))

pkg/generated/clientset/versioned/fake/register.go
54-	v1.AddToGroupVersion(scheme, schema.GroupVersion{Version: "v1"})
55:	utilruntime.Must(AddToScheme(scheme))

cmd/controller/app/server.go
37-func init() {
38:	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
39-
40:	utilruntime.Must(schedulingv1a1.AddToScheme(scheme))

apis/config/scheme/scheme.go
44-func AddToScheme(scheme *runtime.Scheme) {
45:	utilruntime.Must(config.AddToScheme(scheme))
46:	utilruntime.Must(v1.AddToScheme(scheme))
47:	utilruntime.Must(v1beta2.AddToScheme(scheme))
48:	utilruntime.Must(v1beta3.AddToScheme(scheme))

apis/scheduling/scheme/scheme.go
31-func AddToScheme(scheme *runtime.Scheme) {
32:	utilruntime.Must(schedv1alpha1.AddToScheme(scheme
```